tests/components/smappee/test_config_flow.py

```diff
@@ -1,16 +1,326 @@
-"""Test the Smappee config flow."""
-from homeassistant import config_entries, setup
-from homeassistant.components.smappee.const import AUTHORIZE_URL, DOMAIN, TOKEN_URL
+"""Test the Smappee component config flow module."""
+from homeassistant import data_entry_flow, setup
+from homeassistant.components.smappee.const import (
+    CONF_HOSTNAME,
+    CONF_SERIALNUMBER,
+    DOMAIN,
+    ENV_CLOUD,
+    ENV_LOCAL,
+    TOKEN_URL,
+)
+from homeassistant.config_entries import SOURCE_USER, SOURCE_ZEROCONF
 from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
 from homeassistant.helpers import config_entry_oauth2_flow
 
 from tests.async_mock import patch
+from tests.common import MockConfigEntry
 
 CLIENT_ID = "1234"
 CLIENT_SECRET = "5678"
 
 
-async def test_full_flow(hass, aiohttp_client, aioclient_mock):
+async def test_show_user_form(hass):
+    """Test that the user set up form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER},
+    )
+
+    assert result["step_id"] == "environment"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_show_user_host_form(hass):
+    """Test that the host form is served after choosing the local option."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER},
+    )
+    assert result["step_id"] == "environment"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"environment": ENV_LOCAL}
+    )
+
+    assert result["step_id"] == ENV_LOCAL
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_show_zeroconf_connection_error_form(hass):
+    """Test that the zeroconf confirmation form is served."""
+    with patch("pysmappee.api.SmappeeLocalApi.logon", return_value=None):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_ZEROCONF},
+            data={
+                "host": "1.2.3.4",
+                "port": 22,
+                CONF_HOSTNAME: "Smappee1006000212.local.",
+                "type": "_ssh._tcp.local.",
+                "name": "Smappee1006000212._ssh._tcp.local.",
+                "properties": {"_raw": {}},
+            },
+        )
+
+        assert result["description_placeholders"] == {CONF_SERIALNUMBER: "1006000212"}
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "zeroconf_confirm"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)